src/megaapi_impl.cpp

```diff
@@ -3202,6 +3344,27 @@ char *MegaApiImpl::getMyXMPPJid()
     return result;
 }
 
+#ifdef ENABLE_CHAT
+char *MegaApiImpl::getMyFingerprint()
+{
+    sdkMutex.lock();
+    if (ISUNDEF(client->me))
+    {
+        sdkMutex.unlock();
+        return NULL;
+    }
+
+    char *result = NULL;
+    if (client->signkey)
+    {
+        result = MegaApi::strdup(client->signkey->genFingerprintHex());
```

Comment:
Memory leak. Since the returned value is new, it could be directly set to "result"